### PR TITLE
Remove Null Access Token Error to enable responses without access tokens

### LINKED ISSUE #3530: ClientAuthError: access_token_entity_null: Access token entity is null, please check logs and cache to ensure a valid access
### Core Library

@azure/msal-browser

### Core Library Version

2.14.1

### Wrapper Library

Not Applicable

### Wrapper Library Version

0

### Description

on handleRedirectPromise() after login

ClientAuthError: access_token_entity_null: Access token entity is null, please check logs and cache to ensure a valid access token is present.

### Error Message

_No response_

### Msal Logs

_No response_

### MSAL Configuration

```javascript
this.app = new Msal.PublicClientApplication({
                auth: {
                    clientId: `${BaseConfig.AppClientID}`,
                    authority: `${BaseConfig.AppLoginAuthority}`,
                    knownAuthorities: [`${BaseConfig.AppLoginAuthority}`],
                    validateAuthority: false,
                    navigateToLoginRequestUrl: true,
                    redirectUri: `${BaseConfig.AppUrl}login`,
                    postLogoutRedirectUri: `${BaseConfig.AppUrl}login`,
                },
                cache: {
                    cacheLocation: "sessionStorage",
                    storeAuthStateInCookie: true,
                    secureCookies: true
                }
            });
```

### Relevant Code Snippets

```javascript
this.app.handleRedirectPromise().then(response => {
                console.log("response", response);
                if (response) {
                    this.handleResponse(response);
                }
            }).catch(error => {
                console.log(error);
            });
```

### Reproduction Steps

1. immediately after login

### Expected Behavior

should return object with correct data.

### Identity Provider

Azure B2C Basic Policy

### Browsers Affected (Select all that apply)

Edge

### Regression

2.13.1

### Source

External (Customer)

### PR DESCRIPTION
**Pull request by hectormmg**
This PR:
- Removes the Null Access Token Entity Error thrown in `generateAuthenticationResult` when the cached authentication result contains no access token
- Reverts `AuthenticationResult.expiresOn` type to `Date | null` to support cases in which an AuthenticationResult does not contain an access token
- Fixes #3530

### CODE REVIEW

File: lib/msal-common/src/response/ResponseHandler.ts [hunk 1]
```diff
@@ -319,7 +320,7 @@ export class ResponseHandler {
             expiresOn = new Date(Number(cacheRecord.accessToken.expiresOn) * 1000);
             extExpiresOn = new Date(Number(cacheRecord.accessToken.extendedExpiresOn) * 1000);
         } else {
-            throw ClientAuthError.createAccessTokenEntityNullError();
+            expiresOn = new Date(0);
```
Comment (jasonnutter) [inline lib/msal-common/src/response/ResponseHandler.ts:323]: Please add a logger warning message here.

Comment (hectormmg) [reply lib/msal-common/src/response/ResponseHandler.ts]: @jasonnutter this is a static method in ResponseHandler. The logger isn't directly accessible. I'm concerned the code change necessary will turn this into a bigger PR.  Although I don't mind doing it at all, do we think the overhead is worth it?

**Review by tnorling — APPROVED** [review]

**Review by jasonnutter — APPROVED** [review]

File: change/@azure-msal-common-1fb3e3ef-1b48-47f0-a752-576d0e2a5c1d.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
Comment (jasonnutter) [inline change/@azure-msal-common-1fb3e3ef-1b48-47f0-a752-576d0e2a5c1d.json:2]:
```suggestion
  "type": "patch",
```

File: lib/msal-common/src/response/ResponseHandler.ts (hunk 1 above)
Comment (jasonnutter) [reply lib/msal-common/src/response/ResponseHandler.ts]: That's fine, can be addressed later.

**Pull request merged**